src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -249,7 +249,21 @@ internal async Task<IEnumerable<RazorCodeAction>> GetCSharpCodeActionsFromLangua
                 return Array.Empty<RazorCodeAction>();
             }
 
-            var newRequest = context.Request with { Range = projectedRange };
+            var newContext = context.Request.Context;
+            if (context.Request.Context is OmniSharpVSCodeActionContext omniSharpContext &&
+                omniSharpContext.SelectionRange is not null &&
+                 _documentMappingService.TryMapToProjectedDocumentRange(
+                    context.CodeDocument,
+                    omniSharpContext.SelectionRange,
+                    out var selectionRange))
+            {
+                newContext = omniSharpContext with
+                {
+                    SelectionRange = selectionRange
```

Comment:
If this results in a yellow bar otherwise, I think clearing makes sense.

Reply:
I was hesitant to clear. Theoretically we could set it to the same as the Range and be accurate _most_ of the time.

Reply:
@NTaylorMullen do you know of a scenario that yellow bars? Having a stack would help narrow down expectations

Reply:
> @NTaylorMullen do you know of a scenario that yellow bars? Having a stack would help narrow down expectations

I forget the scenarios in which C# will actually inspect the `SelectionRange`; however, if they do + you're in exp hive + are internal you'll see the yellowbar; otherwise Core Editor suppresses the yellow bar and reports fault telemetry.

Unfortunately I'm not 100% positive on the exact scenario. IIRC @allisonchou and I were able to reproduce easily in default Blazor server app + going to Counter.razor and clicking around the methods / fields there.

Maybe @dibarbet would have a better idea when C# language server inspects selection range?